components/event-processor-manager/org.wso2.carbon.event.processor.manager.core/src/main/java/org/wso2/carbon/event/processor/manager/core/internal/HAManager.java

```diff
@@ -119,6 +118,18 @@ public void tryChangeState() {
         }
     }
 
+    public void verifyState(){
+        if(activeLockAcquired &&
```

Comment:
Thanks @keizer619 , Done in pull request - https://github.com/wso2/carbon-analytics-common/pull/109